modules/feegrant/module.go

```diff
@@ -0,0 +1,36 @@
+package feegrant
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	"github.com/forbole/bdjuno/v2/database"
+
+	"github.com/forbole/juno/v2/modules"
+)
+
+var (
+	_ modules.Module        = &Module{}
+	_ modules.MessageModule = &Module{}
```

Comment:
Inside the `feegrant` module, grants can be removed for three main reasons: 
1. The grant is expired
2. The grantee has used all the granted amount 
3. The granter revokes the grant

In all these cases, the module will emit an event with the following attributes: 
- Type: `feegrant.EventTypeRevokeFeeGrant` (`revoke_feegrant`)
- Attributes:
   - `feegrant.AttributeKeyGranter` with value the address of the granter
   - `feegrant.AttributeKeyGrantee` with value the address of the grantee

To properly handle those cases, we should implement the `BlockModule` interface for this method and leverage the `EndBlock` events list to find all events that signal a grant should be removed. For each even found, we should remove that grant from the database.

Reply:
I added the `BlockModule` and handled the EventTypeRevokeFeeGrant event type, but when I test it on my local, the `EndBlock` array is always empty and it doesn't trigger the method to remove expired grant

Reply:
@MonikaCat I've further investigated into this and I found out it's caused by a flawless logic inside the `feegrant` module of the Cosmos SDK. I've now opened an issue here: https://github.com/cosmos/cosmos-sdk/issues/10704. I think we can keep it like you implemented it right now (using the `EndBlocker`) and then simply fix it once that's solved

Reply:
Ahh okay! In that case let's merge it when you approve the changes 👍🏻